Fix updating of the cursor size

jdujava opened the pull request:
Just implemented the proposed solution in #4807, since there was no activity for a while.

Credit goes to @jen-kosek and @rolandlo.

Fixes #3090, #4807.

rolandlo reviewed (changes requested):
Thanks for the PR. 
The feature works fine. Please remove the unused variable `size` from `createHighlighterOrPenCursor` though.

jdujava commented:
Implemented your suggestion and rebased on master.
The `if` statement whether to draw `STYLUS_CURSOR_BIG` with the current color is now based on the passed `alpha` value.
